crypto/hpke/hpke_util.c

```diff
@@ -173,3 +396,72 @@ EVP_KDF_CTX *ossl_kdf_ctx_create(const char *kdfname, const char *mdname,
     }
     return kctx;
 }
+
+/*
+ * @brief map a string to a HPKE suite based on synonym tables
+ * @param str is the string value
+ * @param suite is the resulting suite
+ * @return 1 for success, otherwise failure
+ */
+int ossl_hpke_str2suite(const char *suitestr, OSSL_HPKE_SUITE *suite)
+{
+    uint16_t kem = 0, kdf = 0, aead = 0;
+    char *st = NULL;
+    char *instrcp = NULL;
+    size_t inplen = 0;
+    int labels = 0;
+    const synonymttab_t *synp = NULL;
+    uint16_t *targ = NULL;
+    size_t i, j;
+    size_t outsize = 0;
+    size_t insize = 0;
+
+    if (suitestr == NULL || suite == NULL)
+        return 0;
+    /* See if it contains a mix of our strings and numbers  */
+    inplen = OPENSSL_strnlen(suitestr, OSSL_HPKE_MAX_SUITESTR);
+    if (inplen >= OSSL_HPKE_MAX_SUITESTR)
+        return 0;
+    instrcp = OPENSSL_strndup(suitestr, inplen);
+    st = strtok(instrcp, ",");
+    if (st == NULL) {
+        OPENSSL_free(instrcp);
+        return 0;
+    }
+    while (st != NULL && ++labels <= 3) {
```

Comment:
This code is currently parsing only the first 3 tokens.
No need to raise an error if more than 3 tokens are present ?

Reply:
Good point yeah, you're right that as of now "0x10,0x01,0x01,blah" will work, which is a little odd, given "0x10,0x01,0x01 blah" will fail, so the flexibility wouldn't be useful really. I think an earlier version would have failed as expected for both strings so I've changed it to do that again (by checking ``st == NULL`` at the end before returning success).